### PR TITLE
CONCF-138 add additional logging to WikiaLocalFileShared

### PR DESCRIPTION
**Pull request by bognix**
Broken video was causing 500 Errors - wrong metadata type (expecting array, stdClass found).
@macbre fixed the video and I added additional logging to gather some information in case situation repeats.

ping @macbre

### CODE REVIEW

File: extensions/wikia/VideoHandlers/filerepo/WikiaLocalFileShared.class.php [hunk 1]
```diff
@@ -289,9 +289,19 @@ function metaValue( $key, $default = '' ) {
 			}
 		}
 
-		if ( !empty($this->metadata[$key]) ) {
-			$value = $this->metadata[$key];
+		if ( is_array( $this->metadata ) ) {
+			if ( !empty($this->metadata[$key]) ) {
```
Comment (macbre) [inline extensions/wikia/VideoHandlers/filerepo/WikiaLocalFileShared.class.php]: Whitespaces :)

**Comment by hakubo** [comment]
LGTM - @macbre how about you?

File: extensions/wikia/VideoHandlers/filerepo/WikiaLocalFileShared.class.php [hunk 2]
```diff
@@ -289,9 +289,19 @@ function metaValue( $key, $default = '' ) {
...
+				$value = $this->metadata[$key];
+			} else {
+				$value = $default;
+			}
 		} else {
+			\Wikia\Logger\WikiaLogger::instance()->error(
+				'File metadata not an instance of an array. Expecting array', [
+					'type' => gettype( $this->metadata ),
+					'file' => $this->oFile
+				]
```
Comment (macbre) [inline extensions/wikia/VideoHandlers/filerepo/WikiaLocalFileShared.class.php]: You can add `'exception' => new Exception()` to log the backtrace for easier debugging.

**Comment by macbre** [comment]
LGTM, please check unit tests

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**